benchexec/tools/cpachecker.py

```diff
@@ -121,23 +142,31 @@ def project_url(self):
         return "https://cpachecker.sosy-lab.org/"
 
     def _get_additional_options(self, existing_options, task, rlimits):
+        # If at least one option uses "--" style, we use that for options added here.
+        # Otherwise use "-" to be safe for old versions of CPAchecker.
+        new_option_style = any(option.startswith("--") for option in existing_options)
```

Comment:
How about providing `_get_additional_options` with the executable, and use that as an additional case (or the only case?) for `new_option-style`?
That would differentiate between the option style based on the implied CPAchecker version and not require a heuristic.

Reply:
You mean based on `executable.endswith("cpa.sh")`? I started with this at first, actually, but then switched to the current style to have command lines in a consistent style even if people use old-style arguments with a new CPAchecker version (which I guess will happen a lot because people have a lot of benchmark definitions that they will continue to use, or take a benchmark definition from last SV-COMP).

I also think the current decision is less of a heuristic than one based on the executable: If there is an argument with `--` in the command line, then we certainly have a CPAchecker version that supports them. But one could in principle cherry-pick the new executable in a branch of CPAchecker without also cherry-picking the new command-line arguments.

Reply:
Ok :-)

Reply:
FTR: https://gitlab.com/sosy-lab/software/cpachecker/-/issues/1218#note_1952602398 is also an argument for the current implementation.